identity/webapp/src/frontend/AccountManagement/DeleteAccount.tsx

```diff
@@ -1,33 +1,93 @@
 import SpacingComponent from '@weco/common/views/components/SpacingComponent/SpacingComponent';
+import { PasswordInput } from '../Shared/PasswordInput';
+import { useRequestDelete } from '../hooks/useRequestDelete';
+import { ErrorMessage } from '../Shared/ErrorMessage';
 
 // TODO: Update this to prod.
 const logo = 'https://identity-public-assets-stage.s3.eu-west-1.amazonaws.com/images/wellcomecollections-150x50.png';
-import styled from 'styled-components';
 
 const LogoContainer = styled.div`
-   {
-    display: flex;
-    width: 200px;
-  }
+  display: flex;
+  width: 200px;
 `;
 
-export const DeleteAccount = () => {
+const DangerButton = styled(SolidButton)`
+  background: ${(props) => props.theme.color('red')};
+  border-color: ${(props) => props.theme.color('red')};
+`;
+
+export type DeleteAccountProps = {
+  onCancel: () => void;
+};
+
+export const DeleteAccount: React.FC<DeleteAccountProps> = ({ onCancel }) => {
+  const [password, setPassword] = useState('');
+  const [isSuccess, setIsSuccess] = useState(false);
+  const [isIncorrectPassword, setIsIncorrectPassword] = useState(false);
+  const [requestDelete] = useRequestDelete();
+
+  const handlePasswordChange = (enteredValue: string) => {
+    setIsIncorrectPassword(false);
+    setPassword(enteredValue);
+  };
+
+  const handleSuccess = () => setIsSuccess(true);
+  const handleFailure = (statusCode?: number) => {
+    switch (statusCode) {
+      case 401: {
+        setIsIncorrectPassword(true);
+        setPassword('');
+        break;
+      }
+      default: {
+        console.error('Error');
+      }
+    }
+  };
+
+  const handleConfirmDelete = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    requestDelete({ password }, handleSuccess, handleFailure);
+  };
+
   return (
     <div>
       <LogoContainer>
         <img src={logo} alt="Wellcome Collection Logo" height="200px" />
       </LogoContainer>
       <SpacingComponent />
-
       <h1 className="font-wb font-size-1">Delete Account</h1>
       <SpacingComponent />
-      <p>Are you sure you want to delete your account?</p>
-      <SolidButton>Yes, delete my account</SolidButton>
-      <SpacingComponent />
-      <a href="TBC">Cancel</a>
+      {isSuccess ? (
+        <>
+          <p>Your request for account deletion has been received.</p>
+          <p>
+            Our Library enquiries team will now progress your request. If there are any issues they will be in touch
+            otherwise your account will be removed.
+          </p>
+        </>
+      ) : (
+        <>
+          <p>Are you sure you want to delete your account?</p>
+          <p>To permanently delete your account please enter your password and confirm.</p>
+          <form onSubmit={handleConfirmDelete}>
+            <PasswordInput label="Password" id="password" value={password} setValue={handlePasswordChange} />
+            {isIncorrectPassword && <ErrorMessage>Incorrect password</ErrorMessage>}
```

Comment:
Very minor nitpick, but I think this would be a little bit clearer if it used a ternary like:

```
{isIncorrectPassword ? <ErrorMessage>Incorrect password</ErrorMessage> : null}
```

Reply:
`&&` here is pretty standard. It's specifically mentioned in the React docs: 

https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Reply:
Yeah, I'm happy for this to be left as is. My concern was about deliberately relying on Javascript's odd behaviour around truthy values given how much discussion has been generated on slack lately around people not understanding it. I think in this case it's clear enough what the intent is that it's not a problem.

Reply:
Fair comment, there has been a lot of confusion.

In this case, a variable prefixed `is` is almost certainly a boolean (the editor's TypeScript integration should verify this), so `boolean && <OtherThing />` can only ever be `<OtherThing />` or `false`